### PR TITLE
Update title.service.ts

### PR DESCRIPTION
**Pull request by DmitriyPotapov**
## What: Maybe in this case it is worth unsubscribing from observable?

Issue number: N/A

### CODE REVIEW

**Comment by timdeschryver** [comment]
I don't think this is needed because the the service exists as long as the application's life cycle.
Thoughts @tomastrajan ?

File: src/app/core/title/title.service.ts [hunk 1]
```diff
@@ -27,8 +28,12 @@ export class TitleService {
       translate
         .get(title)
         .pipe(filter(translatedTitle => translatedTitle !== title))
-        .subscribe(translatedTitle =>
-          this.title.setTitle(`${translatedTitle} - ${env.appName}`)
+        .takeUntil(this.unsubscribe$)
```
Comment (timdeschryver) [inline src/app/core/title/title.service.ts]: You'll have to use the pipeable operator here.

```
 .pipe(filter(translatedTitle => translatedTitle !== title), takeUntil(this.unsubscribe$)
)
```

File: src/app/core/title/title.service.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { environment as env } from '@env/environment';
 
 @Injectable()
 export class TitleService {
+  private unsubscribe$ = new Subject();
```
Comment (timdeschryver) [inline src/app/core/title/title.service.ts]: This isn't doing anything now.

In order to unsubscribe the observables, you will have to emit (and complete) the Subject's stream when the service is destroyed.

```
this.unsubscribe$.next();
this.unsubscribe$.complete();
```

**Comment by tomastrajan** [comment]
@DmitriyPotapov I agree with the @timdeschryver  here, the service exists for the whole duration of the application life and so there is no reason to make things more complicated. Thanks for your effort and hope you enjoy the starter project!

**Pull request closed without merging**